pkg/controller/daemon/daemon_controller.go

```diff
@@ -1210,10 +1297,25 @@ func (dsc *DaemonSetsController) syncDaemonSet(ctx context.Context, key string)
 	if err != nil {
 		return err
 	}
+
+	dsNamespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+
+	// If our writes have not yet been observed by our informers, we should not
+	// continue since our reads will not be in sync with our previously enacted
+	// state.
+	if !dsc.consistencyStore.IsReady(dsNamespacedName) {
+		logger.Info("Daemonset is not ready for reconciliation, previous writes have not been observed in the cache", "ds", dsNamespacedName)
```

Comment:
I think we may want to switch from `IsReady(namespacedName) bool` to `EnsureReady(namespacedName) error` so we can return details about the group/resource/expectedRV/actualRV that blocked readiness and log it here